### PR TITLE
Flake8 improvements

### PR DESCRIPTION
**Pull request by VitaliyaIoffe**
This is my suggestion about using flake8. Currently mkrepo contains a small group of problematic errors, so, I'd like to fix it and set the flake8 as a mandatory workflow.
 All changes are discussable, please, write your preferences in a code style.

### CODE REVIEW

File: debrepo.py [hunk 1]
```diff
@@ -151,7 +152,8 @@ def __init__(self, component='main', arch='amd64'):
         self.arch = arch
 
     def parse_deb(self, debfile):
-        if subprocess.call('ar t ' + debfile + ' | grep control.tar.gz', shell=True) == 0:
+        if subprocess.call('ar t ' + debfile + ' | grep control.tar.gz',
```
Comment (LeonidVas) [inline debrepo.py:155]: I suggest using line width <= 100 instead of 80 as default.

File: debrepo.py [hunk 2]
```diff
@@ -498,7 +505,7 @@ def process_index_file(repo_info, path, dist, component, arch, index_type):
     elif index_type == 'sources':
         index = SourceIndex()
     else:
-        raise(RuntimeError('Unknown index type: ' + index_type))
+        raise (RuntimeError('Unknown index type: ' + index_type))
```
Comment (LeonidVas) [inline debrepo.py:508]: What is this error?(In PEP8)

File: debrepo.py [hunk 3]
```diff
@@ -741,8 +750,8 @@ def update_index_files(repo_info, index_type):
             for checksum_type in ['md5', 'sha1', 'sha256']:
                 h = hashlib.new(checksum_type)
                 h.update(data)
-
-                repo_info.checksums[dist][(checksum_type, path)] = h.hexdigest()
+                type_path_tuple = (checksum_type, path)
```
Comment (LeonidVas) [inline debrepo.py:753]: What is this error? (In PEP8)

File: .github/workflows/flake8.yml [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+        uses: actions/setup-python@v2
+        with:
+          python-version: '3.9'
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          if [ -f requirements.txt ]; then pip install -r requirements.txt; fi
+      - name: Run flake8
+        run: |
+          flake8 .
```
Comment (LeonidVas) [inline .github/workflows/flake8.yml:22]: I suggest to move the launch of `flake8` to the `Makefile`. As result, to run the lint will be enough to execute `make lint`. It is really very comfortable to perform all operations like test, lint, install ... over the `Makefile.`

File: debrepo.py [hunk 5]
```diff
@@ -1,19 +1,20 @@
 #!/usr/bin/env python
 
+import bz2
```
[the remark this answers is not in the document]
Comment (VitaliyaIoffe) [reply debrepo.py:3]: Pycharm provides a good feature `Optime imports, which is reordering imports with python rules. I didn't know abput `flake8-isort` - have to look deeply.

Good point about additional commit

[the remark this answers is not in the document]
Comment (Totktonada) [reply debrepo.py:3]: If you want to tweak imports order, add automation that will ensure that the correct order is kept.

File: requirements.txt [hunk 6]
```diff
@@ -1 +1,2 @@
 boto3==1.4.1
+flake8>=3.7.9
```
Comment (Totktonada) [inline requirements.txt:2]: flake8 is not necessary for the module itself. It is testing requirement. Plase, move to `requirements-test.txt`.

**Comment by knazarov** [comment]
Is this ready to be merged?

**Comment by VitaliyaIoffe** [comment]
> Is this ready to be merged?

No, still in progress.

File: debrepo.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (VitaliyaIoffe) [reply debrepo.py:3]: Use `isort` for sorting imports

File: debrepo.py (hunk 1 above)
Comment (VitaliyaIoffe) [reply debrepo.py]: Use 

```
max-line-length = 100
max-doc-length = 100
```

File: debrepo.py (hunk 2 above)
Comment (VitaliyaIoffe) [reply debrepo.py]: Remove to the commit with refactoring

File: debrepo.py (hunk 3 above)
Comment (VitaliyaIoffe) [reply debrepo.py]: Removed this change

File: .github/workflows/flake8.yml (hunk 4 above)
Comment (VitaliyaIoffe) [reply .github/workflows/flake8.yml]: add target `lint`

File: debrepo.py [hunk 7]
```diff
@@ -613,7 +621,7 @@ def process_index_units(repo_info, tempdir, index_type, force=False):
         expr = r'^.*\.dsc$'
         tmp_filename = 'source.dsc'
     else:
-        raise(RuntimeError('Unknown index type: ' + index_type))
+        raise (RuntimeError('Unknown index type: ' + index_type))
```
[the remark this answers is not in the document]
Comment (VitaliyaIoffe) [reply debrepo.py]: Fixed

File: requirements.txt (hunk 6 above)
Comment (VitaliyaIoffe) [reply requirements.txt]: add requirements-test.txt

File: test/test_debrepo.py [hunk 8]
```diff
@@ -4,22 +4,26 @@
 import sys
 import unittest
 
-sys.path.insert(1, os.path.join(sys.path[0], '..'))
 import debrepo
 
+sys.path.insert(1, os.path.join(sys.path[0], '..'))
```
[the remark this answers is not in the document]
Comment (VitaliyaIoffe) [reply test/test_debrepo.py]: it does not refer to this pr more

**Review by LeonidVas — APPROVED** [review]
LGTM.

File: Makefile [hunk 9]
```diff
@@ -1,6 +1,14 @@
 index.html : README.md static/skeleton.css static/box.svg static/template.html
 	pandoc -s README.md -o index.html -c static/skeleton.css --template static/template.html -T mkrepo
 
+lint: flake8 isort
+
+flake8:
+	flake8 .
+
+isort:
+	isort . -c
```
Comment (Totktonada) [inline Makefile:10]: Nit: Please, mark them as PHONY targets.

**Review by Totktonada — APPROVED** [review]
LGTM.

File: Makefile (hunk 9 above)
Comment (VitaliyaIoffe) [reply Makefile]: Add PHONY to all new targets

**Pull request merged**